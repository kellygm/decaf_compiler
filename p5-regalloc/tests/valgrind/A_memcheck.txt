==2923175== Memcheck, a memory error detector
==2923175== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==2923175== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==2923175== Command: ../decaf inputs/sanity.decaf
==2923175== 
main:
  push BP
  i2i SP => BP
  addI SP, 0 => SP
  loadI 4 => R0
  i2i R0 => RET
  jump l0
l0:
  i2i BP => SP
  pop BP
  return
RETURN VALUE = 4
==2923175== 
==2923175== HEAP SUMMARY:
==2923175==     in use at exit: 0 bytes in 0 blocks
==2923175==   total heap usage: 1,805 allocs, 1,805 frees, 462,516 bytes allocated
==2923175== 
==2923175== All heap blocks were freed -- no leaks are possible
==2923175== 
==2923175== For lists of detected and suppressed errors, rerun with: -s
==2923175== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
