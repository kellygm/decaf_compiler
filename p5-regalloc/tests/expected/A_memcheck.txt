main:
  push BP
  i2i SP => BP
  addI SP, 0 => SP
  loadI 4 => R0
  i2i R0 => RET
  jump l0
l0:
  i2i BP => SP
  pop BP
  return
RETURN VALUE = 4
